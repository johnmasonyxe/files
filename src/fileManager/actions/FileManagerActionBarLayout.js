// @flow
import React from 'react';
import FileManagerActionBarUpload from './FileManagerActionBarUpload';
import FileManagerActionBarSearch from './FileManagerActionBarSearch';
import './actions.scss';

type Props = {
    uploadFile: any => void,
    setSearchText: string => void,
}
const FileManagerActionBarLayout = (props: Props) => {
    const {
        uploadFile,
        setSearchText,
    } = props;

    return (
        <div className='actions-layout-container'>
            <FileManagerActionBarSearch setSearchText={setSearchText}/>
            <div className='actions-layout-upload-container'>
                <FileManagerActionBarUpload uploadFile={uploadFile}/>
            </div>
        </div>
    );
};

export default FileManagerActionBarLayout;
