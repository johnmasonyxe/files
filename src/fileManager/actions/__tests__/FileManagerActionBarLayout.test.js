// @flow
import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FileManagerActionBarLayout from '../FileManagerActionBarLayout';

Enzyme.configure({adapter: new Adapter()});

describe('FileManagerActionBarLayout', () => {

    // some people prefer to test things this way, i find it too fragile
    it('should render correctly', () => {
        const props = {
            setSearchText: jest.fn(),
            uploadFile: jest.fn(),
        };

        const component = renderer.create(
            <FileManagerActionBarLayout {...props}/>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // if testing components is a priority/requirement this is how i would go about it
    // if you ensure all of the right components are rendered
    //   and you test the logic in those components
    //   and you trust that flow/typescript is ensuring you pass the correct props around
    //   you should be good to go
    it('should render the right components', () => {
        const props = {
            setSearchText: jest.fn(),
            uploadFile: jest.fn(),
        };
        const actionBar = mount(<FileManagerActionBarLayout{...props} />);
        expect(actionBar.find('FileManagerActionBarSearch').length).toBe(1);
        expect(actionBar.find('FileManagerActionBarUpload').length).toBe(1);
    });
});
