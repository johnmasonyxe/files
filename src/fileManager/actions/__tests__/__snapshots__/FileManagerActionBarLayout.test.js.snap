// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileManagerActionBarLayout should render correctly 1`] = `
<div
  className="actions-layout-container"
>
  <span
    className="ant-input-search actions-search-width ant-input-search-enter-button ant-input-group-wrapper"
  >
    <span
      className="ant-input-wrapper ant-input-group"
    >
      <input
        className="ant-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="search documents"
        style={null}
        type="text"
        value=""
      />
      <span
        className="ant-input-group-addon"
      >
        <button
          className="ant-btn ant-input-search-button ant-btn-primary"
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
        >
          <span
            aria-label="search"
            className="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </span>
  <div
    className="actions-layout-upload-container"
  >
    <span
      className=""
    >
      <div
        className="ant-upload ant-upload-select ant-upload-select-text"
      >
        <span
          className="ant-upload"
          onClick={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <input
            accept=""
            directory={null}
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file"
            webkitdirectory={null}
          />
          <button
            className="ant-btn"
            onClick={[Function]}
            style={
              Object {
                "width": "300px",
              }
            }
            type="button"
          >
            <div>
              <span
                aria-label="upload"
                className="anticon anticon-upload"
                role="img"
                style={
                  Object {
                    "color": undefined,
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="upload"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
               Upload a PNG/JPG
            </div>
          </button>
        </span>
      </div>
    </span>
  </div>
</div>
`;
